scripts/azureml-assets/azureml/assets/validate_assets.py

```diff
@@ -574,6 +595,188 @@ def validate_model_assets(latest_asset_config: assets.AssetConfig, validated_ass
         return 1
 
 
+def validate_model_spec(asset_config: assets.AssetConfig, validated_model_map: dict):
+    """Validate model spec."""
+    model = load_model(asset_config.spec_with_path)
+    if asset_config.type != assets.config.ModelType.MLFLOW:
+        logger.print(
+            f"Bypass validation for {asset_config.name} as model type is: {asset_config.type.value}"
+        )
+        return 0
+
+    # confirm must have tags
+    if not model.tags.get(MLFlowModelTags.TASK):
+        _log_error(asset_config.file_name_with_path, f"{MLFlowModelTags.TASK} missing")
+        return 1

```

Comment:
Would it be possible to use something like error_count that many of the other validation methods are using? If the task tag is missing, this function would return that there's 1 error and not consider the rest of the checks in this function

Reply:
updated. Validating few scenarios